src/lib/y2packager/product.rb

```diff
@@ -0,0 +1,169 @@
+      @display_name = display_name
+      @version = version
+      @arch = arch.to_sym if arch
+      @status = status.to_sym if status
+      @category = category.to_sym if category
+      @vendor = vendor
+      @order = order
+      @installation_package = installation_package
+    end
+
+    # Compare two different products
+    #
+    # If arch, name, version and vendor match they are considered the
+    # same product.
+    #
+    # @return [Boolean] true if both products are the same; false otherwise
+    def ==(other)
+      result = arch == other.arch && name == other.name &&
+        version == other.version && vendor == other.vendor
+      log.info("Comparing products: '#{arch}' <=> '#{other.arch}', '#{name}' <=> '#{other.name}', "\
+        "'#{version}' <=> '#{other.version}', '#{vendor}' <=> '#{other.vendor}' => "\
+        "result: #{result}")
+      result
+    end
+
+    # is the product selected to install?
+    #
+    # @return [Boolean] true if it is selected
+    def selected?
+      Yast::Pkg.ResolvableProperties(name, :product, "").any? do |res|
+        res["status"] == :selected
+      end
+    end
+
+    # select the product to install
+    #
+    # Only the 'name' will be used to select the product, ignoring the
+    # architecture, version, vendor or any other property. libzypp will take
+    # care of selecting the proper product.
+    #
+    # @return [Boolean] true if the product has been sucessfully selected
+    def select
+      log.info "Selecting product #{name} to install"
+      Yast::Pkg.ResolvableInstall(name, :product, "")
+    end
+
+    def restore
+      log.info "Restoring product #{name} status"
+      Yast::Pkg.ResolvableNeutral(name, :product, true)
+    end
+
+    # Return a package label
+    #
+    # It will use 'display_name', 'short_name' or 'name'.
+    #
+    # @return [String] Package label
+    def label
+      display_name || short_name || name
+    end
+
+    # Return the license to confirm
+    #
+    # It will return the empty string ("") if the license does not exist or if
+    # it was already confirmed.
+    #
+    # @return [String,nil] Product's license; nil if the product was not found.
+    def license(lang = nil)
+      license_lang = lang || Yast::Language.language
+      Yast::Pkg.PrdGetLicenseToConfirm(name, license_lang)
+    end
+
+    # Determines whether the product has a license
+    #
+    # @return [Boolean] true if the product has a license
+    def license?
+      return false unless license
+      license != ""
+    end
+
+    # Determine whether the license should be accepted or not
+    #
+    # @return [Boolean] true if the license acceptance is required
+    def license_confirmation_required?
+      Yast::Pkg.PrdNeedToAcceptLicense(name)
+    end
+
+    # Confirm the license for the product
+    def confirm_license
+      Yast::Pkg.PrdMarkLicenseConfirmed(name)
+    end
+
+    # Unconfirm the license for the product
+    def unconfirm_license
```

Comment:
You're right. It's already addressed on #267. I followed your suggestion and replaced confirm/unconfirm with a single method which receives a boolean.